src/main/java/mate/academy/bookstore/service/CategoryServiceImpl.java

```
@@ -0,0 +1,52 @@
+package mate.academy.bookstore.service;
+
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookstore.dto.category.CategoryDto;
+import mate.academy.bookstore.exception.EntityNotFoundException;
+import mate.academy.bookstore.mapper.CategoryMapper;
+import mate.academy.bookstore.model.Category;
+import mate.academy.bookstore.repository.CategoryRepository;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CategoryServiceImpl implements CategoryService {
+    private final CategoryRepository categoryRepository;
+    private final CategoryMapper categoryMapper;
+
+    @Override
+    public List<CategoryDto> findAll(Pageable pageable) {
+        return categoryRepository.findAll().stream()
+                .map(categoryMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public CategoryDto getById(Long id) {
+        Category category = categoryRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Can't get category by id " + id));
+        return categoryMapper.toDto(category);
+    }
+
+    @Override
+    public CategoryDto save(CategoryDto categoryDto) {
+        Category category = categoryMapper.toModel(categoryDto);
+        return categoryMapper.toDto(categoryRepository.save(category));
+    }
+
+    @Override
+    public CategoryDto update(Long id, CategoryDto categoryDto) {
+        Category category = categoryMapper.toModel(categoryDto);
+        category.setId(id);
+        return categoryMapper.toDto(categoryRepository.save(category));
+    }
+
+    @Override
+    public void deleteById(Long id) {
+        if (categoryRepository.findById(id).isPresent()) {
```

let's throw an exception if category not exists